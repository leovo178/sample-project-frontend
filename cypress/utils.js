/* eslint-disable no-undef */
const generateRandomNumber = (min, max) => {
	min = Math.ceil(min)
	max = Math.floor(max)
	return Math.floor(Math.random() * (max - min)) + min
}

const clickItemByTestId = testId =>
	cy
		.findByTestId(testId)
		.should('exist')
		.click()

const mockMessage = 'mockmessage'

export { generateRandomNumber, clickItemByTestId, mockMessage }
