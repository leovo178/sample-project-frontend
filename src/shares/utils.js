const AUTH_TOKEN = 'AUTH_TOKEN'

const getToken = () => localStorage.getItem(AUTH_TOKEN)
const setToken = token => localStorage.setItem(AUTH_TOKEN, token)
const deleteToken = () => localStorage.removeItem(AUTH_TOKEN)

const validateEmail = email => {
	const emailRegex = /^[a-z][a-z0-9_\.]{4,32}@[a-z0-9]{2,}(\.[a-z0-9]{2,4}){1,2}$/gm
	return email.match(emailRegex)
}

const validateName = name => {
	const nameRegex = /^[a-zA-Z0-9]{3,30}$/
	return name.match(nameRegex)
}

const validatePassword = password => {
	const passwordRegex = /^[a-zA-Z0-9]{3,}$/
	return password.match(passwordRegex)
}

export {
	getToken,
	setToken,
	deleteToken,
	validateEmail,
	validateName,
	validatePassword,
}
