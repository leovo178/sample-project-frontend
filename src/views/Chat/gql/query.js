import gql from 'graphql-tag'

const GET_SELECTED_USER_OF_CHAT = gql`
	query {
		selectedUserOfChat @client {
			id
			displayName
			avatar
		}
	}
`
const GET_DRAFT_LIST = gql`
	query {
		draftList @client {
			items {
				toInterestId
				message
			}
		}
	}
`

const GET_USER_INFO = gql`
	query {
		me {
			id
			name
			email
			image {
				link
				filename
			}
			followers {
				items {
					displayName
					id
					avatar
				}
			}
		}
	}
`

const GET_NEW_NOTI_MESSAGE_LIST = gql`
	query {
		newNotiMessageList @client {
			items {
				fromInterestedId
				numberNoti
				lastMessage
			}
		}
	}
`

const GET_ZALO_MESSAGE_LIST = gql`
	query ZaloMessageList($query: ZaloMessageListInput) {
		zaloMessageList(query: $query) {
			items {
				id
				content
				from {
					displayName
					id
					avatar
				}
				to {
					displayName
					id
					avatar
				}
				timestamp
			}
			hasNext
			total
		}
	}
`

const GET_USER_SEARCH_TEXT = gql`
	query {
		userSearchValueOfChat @client
	}
`

export {
	GET_SELECTED_USER_OF_CHAT,
	GET_DRAFT_LIST,
	GET_ZALO_MESSAGE_LIST,
	GET_USER_SEARCH_TEXT,
	GET_NEW_NOTI_MESSAGE_LIST,
	GET_USER_INFO,
}
