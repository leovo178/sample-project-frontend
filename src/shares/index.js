import { getToken, setToken, deleteToken } from './utils'

export { getToken, setToken, deleteToken }
