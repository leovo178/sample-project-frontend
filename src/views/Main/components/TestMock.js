import React from 'react'
import { useQuery } from '@apollo/react-hooks'
import { MESSAGE_LIST } from '@views/Message/gql/query'
import { PAGE_LIMIT } from '@src/configs.local'
import { GET_USER_SEARCH_TEXT } from '@views/User/gql/query'

const TestMock = () => {
	const { data } = useQuery(MESSAGE_LIST, {
		variables: { userId: 'test', limit: PAGE_LIMIT },
		// fetchPolicy: 'network-only','messageList'
		// notifyOnNetworkStatusChange: true,
	})
	console.log('data message', data)

	const {
		data: { userSearchValue },
	} = useQuery(GET_USER_SEARCH_TEXT)

	console.log('userSearchValue', userSearchValue)

	return <div>hihi</div>
}

export default TestMock
