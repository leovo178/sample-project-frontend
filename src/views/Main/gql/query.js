import gql from 'graphql-tag'

import { USER } from './fragment'

const GET_SELECTED_USER_OF_MAIN = gql`
	query {
		selectedUserOfMain @client {
			...UserOfMain
		}
	}
	${USER}
`

const GET_USER_SEARCH_TEXT = gql`
	query {
		userSearchValueOfMain @client
	}
`

export { GET_SELECTED_USER_OF_MAIN, GET_USER_SEARCH_TEXT }
