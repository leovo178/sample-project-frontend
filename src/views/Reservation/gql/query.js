import gql from 'graphql-tag'

const GET_RESERVATION_QUEUE = gql`
	query {
		reservationQueue @client {
			items {
				id
				type
				patient
				doctor
				time
				unixTime
			}
		}
	}
`

export { GET_RESERVATION_QUEUE }
