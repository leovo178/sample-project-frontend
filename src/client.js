import { ApolloClient } from 'apollo-client'
import { createUploadLink } from 'apollo-upload-client'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { setContext } from 'apollo-link-context'

import { getToken } from './shares/utils'
import { APOLLO_SERVER } from './configs.local'

const typeDefs = {}

const resolvers = {
	Mutation: {
		setUserSearchValue: (_, { searchValue }, { cache }) => {
			cache.writeData({
				data: {
					userSearchValue: searchValue,
				},
			})
			return searchValue
		},
		setSelectedUser: (_, { selectedUser }, { cache }) => {
			cache.writeData({
				data: {
					selectedUser,
				},
			})
			return selectedUser
		},
		setUploadedFile: (_, { file }, { cache }) => {
			if (file) {
				cache.writeData({
					data: {
						file,
					},
				})
			}
			return file
		},
		setUserSearchValueOfMain: (_, { searchValue }, { cache }) => {
			cache.writeData({
				data: {
					userSearchValueOfMain: searchValue,
				},
			})
			return searchValue
		},
		setSelectedUserOfMain: (_, { selectedUser }, { cache }) => {
			cache.writeData({
				data: {
					selectedUserOfMain: selectedUser,
				},
			})
			return selectedUser
		},

		setMessageSearchValueOfMain: (_, { searchValue }, { cache }) => {
			cache.writeData({
				data: {
					messageSearchValueOfMessage: searchValue,
				},
			})
			return searchValue
		},
		setMessageCreateValueOfMain: (_, { createValue }, { cache }) => {
			cache.writeData({
				data: {
					messageCreateValueOfMessage: createValue,
				},
			})
			return createValue
		},
		resetCache: (_, { data }, { cache }) => {
			cache.writeData({
				data,
			})
			return data
		},
	},
}

const httpLink = createUploadLink({
	uri: APOLLO_SERVER,
	credentials: 'same-origin',
})

const authLink = setContext((_, { headers }) => {
	const token = getToken()
	return {
		headers: {
			...headers,
			authorization: token ? `Bearer ${token}` : '',
		},
	}
})

const cache = new InMemoryCache()

const client = new ApolloClient({
	cache,
	link: authLink.concat(httpLink),
	typeDefs,
	resolvers,
})

const initialState = {
	userSearchValue: '',
	messageCreateValueOfMessage: '',
	messageSearchValueOfMessage: '',
	userSearchValueOfMain: '',
	selectedUser: {
		id: '',
		name: '',
		email: '',
		__typename: 'User',
	},
	file: {
		filename: '',
		link: '',
		__typename: 'File',
	},
	selectedUserOfMain: {
		id: '',
		name: '',
		email: '',
		__typename: 'UserOfMain',
	},
}

cache.writeData({
	data: initialState,
})

export { client, initialState }
