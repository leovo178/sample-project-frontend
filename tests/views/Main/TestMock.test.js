import React from 'react'
import { act, cleanup, fireEvent, render } from '@testing-library/react'
import { createMockClient } from 'mock-apollo-client'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { initialState, resolvers } from '@src/client'
import { renderWithApolloClient } from '@tests/shares/utils'
import TestMock from '../../../src/views/Main/components/TestMock'
import { MESSAGE_LIST } from '@views/Message/gql/query'
import { MockedProvider } from '@apollo/react-testing'
import { mount, ReactWrapper } from 'enzyme';

describe('TestMock', () => {
	afterEach(() => {
		cleanup()
	})

	const cache = new InMemoryCache()
	cache.writeData({
		data: { ...initialState, userSearchValue: 'test_defauilt' },
	})
	const mockClient = createMockClient({
		cache,
		resolvers,
	})

	it('should match snapshot', () => {
		const dogMock = {
			request: {
				query: MESSAGE_LIST,
				variables: { query: { userId: 'testId' } },
			},
			result: {
				data: {
					messageList: {
						items: [{ id: 1, content: 'Buck', lastModified: 'poodle' }],
						hasNext: false,
						total: 1,
					},
				},
			},
		}

		mockClient.setRequestHandler(MESSAGE_LIST, () =>
			Promise.resolve({
				data: {
					messageList: {
						items: [{ id: 1, content: 'Buck', lastModified: 'poodle' }],
						hasNext: false,
						total: 1,
					},
				},
			})
		)

		const { container } = renderWithApolloClient(
			// <MockedProvider mocks={[dogMock]} addTypename={false}>
			<TestMock />,
			// </MockedProvider>,
			mockClient
		)
		expect(container).toMatchSnapshot()
	})
})
