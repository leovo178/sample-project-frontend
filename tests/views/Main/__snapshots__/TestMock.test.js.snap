// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestMock should match snapshot 1`] = `
<div>
  <div>
    hihi
  </div>
</div>
`;
